cmd/argoexec/commands/artifact/delete.go

```
@@ -52,63 +53,121 @@ func NewArtifactDeleteCommand() *cobra.Command {
-					if err != nil {
-						errString := err.Error()
-						artResultNodeStatus.ArtifactResults[artifact.Name] = v1alpha1.ArtifactResult{Name: artifact.Name, Success: false, Error: &errString}
-						return false, err
-					}
-					artResultNodeStatus.ArtifactResults[artifact.Name] = v1alpha1.ArtifactResult{Name: artifact.Name, Success: true, Error: nil}
-					return true, err
-				})
 			}
+		}
+	}
+	return nil
+}
 
-			task.Status.ArtifactResultsByNode[nodeName] = artResultNodeStatus
+func deleteWorker(ctx context.Context, taskQueue chan *request, responseQueue chan response) {
+	for {
+		item, ok := <-taskQueue
+		if !ok {
+			// Done
+			responseQueue <- response{Task: nil, NodeName: "", Err: nil}
+			return
 		}
-		patch, err := json.Marshal(map[string]interface{}{"status": v1alpha1.ArtifactGCStatus{ArtifactResultsByNode: task.Status.ArtifactResultsByNode}})
-		if err != nil {
-			return err
+		var archiveLocation *v1alpha1.ArtifactLocation
+		results := make(map[string]v1alpha1.ArtifactResult)
+		if item.ArtifactNodeSpec.ArchiveLocation != nil {
+			archiveLocation = item.ArtifactNodeSpec.ArchiveLocation
 		}
-		_, err = artifactGCTaskInterface.Patch(context.Background(), task.Name, types.MergePatchType, patch, metav1.PatchOptions{}, "status")
-		if err != nil {
-			return err
+
+		var resources resources
+		resources.Files = make(map[string][]byte) // same resources for every artifact
+		for _, artifact := range item.ArtifactNodeSpec.Artifacts {
+			if archiveLocation != nil {
+				err := artifact.Relocate(archiveLocation)
+				if err != nil {
+					responseQueue <- response{Task: item.Task, NodeName: item.NodeName, Results: results, Err: err}
+					continue
+				}
+			}
+			drv, err := executor.NewDriver(ctx, &artifact, resources)
+			if err != nil {
+				responseQueue <- response{Task: item.Task, NodeName: item.NodeName, Results: results, Err: err}
+				continue
+			}
+
+			err = waitutil.Backoff(retry.DefaultRetry, func() (bool, error) {
```

As @juliev0 states, the err is not handled